service/resource/configmap/desired_test.go

```diff
@@ -1 +1,329 @@
 package configmap
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+	"gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/pkg/api/v1"
+
+	"github.com/giantswarm/micrologger/microloggertest"
+
+	"github.com/giantswarm/prometheus-config-controller/service/prometheus"
+)
+
+func Test_Resource_ConfigMap_GetDesiredState(t *testing.T) {
+	configMapKey := "prometheus.yml"
+	configMapName := "prometheus"
+	configMapNamespace := "monitoring"
+
+	tests := []struct {
+		setUpPrometheusConfiguration *config.Config
```

Comment:
While I went with a `setUp` function in other PRs, I found it got quite cumbersome here. Having more specific test cases, with some more advanced test code helped.